common/components/common/selection/SelectorCollapsible.jsx

```diff
@@ -160,20 +162,24 @@ class SelectorCollapsible<T> extends React.PureComponent<Props<T>, State> {
     return (
       <span>
         <label className="CollapsibleMenuItem">
-          <input type="checkbox" className="ContextualCollapsible-selectAll" onChange={this._selectAll(category)}>
+          <input type="checkbox" className="ContextualCollapsible-selectAll" checked={this.state.isAllChecked} onChange={this._handleSelectAll(category)}>
           </input>Select All
         </label>
       </span>
     )
   }
-  _selectAll(category): React$Node {
-    //get list of tags
-      category.map(function(tag) {
-       return tag.tag_name
-    })
-    //find a way to access list of selected tags in TagSelectorCollapsible
-    // if tag in the above map isn't selected, select it, and then check the Select All box
-    // otherwise remove any tags from the above map from list of selected tags and uncheck the box
+
+  _handleSelectAll(category): React$Node {
```

Comment:
For background, the parent component is `TagSelectorCollapsible` which passes a lot of props to `SelectorCollapsible`  -- I'm trying to work sort of in parallel to how the tag selection/unselection works as is.

This code, which doesn't work, is supposed to respond to someone clicking the select all box by first checking whether or not it's checked (looking at this.state.isAllChecked) and if the box wasn't checked, run the code to check all the boxes in a given category. If it was checked (and the user is unchecking it) run the code to uncheck all the boxes in a given category.

This is a little tricky because `this.selectOption` (which goes up to `SelectOption()` in TagSelectorCollapsible) will automatically check an unchecked box or uncheck a checked one - but in this case since I want to check everything or uncheck everything I have to filter what I provide to that function, otherwise someone with half the boxes checked in a category would just flip which half was selected/unselected.

Well, that's the theory anyway.